### ISSUE TITLE
Was working, after down/up now getting Challenge did not pass

### ISSUE THREAD

**Issue by natew** [SteveLTN/https-portal#198]
I got https-portal working, using [exactly this configuration](https://github.com/verdaccio/docker-examples/tree/master/https-portal-example).

I changed a setting and ran `docker-compose down` and then `docker-compose up` and I see this now:

```
https-portal_1  | Already registered!
https-portal_1  | Creating new order...
https-portal_1  | Order created!
https-portal_1  | Verifying registry.tryorbit.com...
https-portal_1  | Traceback (most recent call last):
https-portal_1  |   File "/bin/acme_tiny", line 197, in <module>
https-portal_1  |     main(sys.argv[1:])
https-portal_1  |   File "/bin/acme_tiny", line 193, in main
https-portal_1  |     signed_crt = get_crt(args.account_key, args.csr, args.acme_dir, log=LOGGER, CA=args.ca, disable_check=args.disable_check, directory_url=args.directory_url, contact=args.contact)
https-portal_1  |   File "/bin/acme_tiny", line 149, in get_crt
https-portal_1  |     raise ValueError("Challenge did not pass for {0}: {1}".format(domain, authorization))
https-portal_1  | ValueError: Challenge did not pass for registry.tryorbit.com: {u'status': u'invalid', u'challenges': [{u'status': u'invalid', u'url': u'https://acme-staging-v02.api.letsencrypt.org/acme/challenge/k8nSNdEk_XT2K7JFBosKNNUIfhUVzrTvKLrGVGgUkwk/317707396', u'token': u'sqId3vKgMh-QZKm-N4LdJCrFyMg-lja9SZmhYGN0izI', u'type': u'tls-alpn-01'}, {u'status': u'invalid', u'validationRecord': [{u'url': u'http://registry.tryorbit.com/.well-known/acme-challenge/4yECewFte-w4NNPkG2b78gOcXvJCwVsW891fnoRGsG0', u'hostname': u'registry.tryorbit.com', u'addressUsed': u'142.93.94.185', u'port': u'80', u'addressesResolved': [u'142.93.94.185']}], u'url': u'https://acme-staging-v02.api.letsencrypt.org/acme/challenge/k8nSNdEk_XT2K7JFBosKNNUIfhUVzrTvKLrGVGgUkwk/317707397', u'token': u'4yECewFte-w4NNPkG2b78gOcXvJCwVsW891fnoRGsG0', u'error': {u'status': 400, u'type': u'urn:ietf:params:acme:error:connection', u'detail': u'Fetching http://registry.tryorbit.com/.well-known/acme-challenge/4yECewFte-w4NNPkG2b78gOcXvJCwVsW891fnoRGsG0: Connection refused'}, u'type': u'http-01'}, {u'status': u'invalid', u'url': u'https://acme-staging-v02.api.letsencrypt.org/acme/challenge/k8nSNdEk_XT2K7JFBosKNNUIfhUVzrTvKLrGVGgUkwk/317707398', u'token': u'S7jbOBd64xeqhbCk_nQQcOeEUP3jOZx-O_RpMK5AYkY', u'type': u'dns-01'}], u'identifier': {u'type': u'dns', u'value': u'registry.tryorbit.com'}, u'expires': u'2019-06-06T23:50:03Z'}
https-portal_1  | ================================================================================
https-portal_1  | Failed to sign registry.tryorbit.com, is DNS set up properly?
https-portal_1  | ================================================================================
https-portal_1  | Failed to obtain certs for registry.tryorbit.com
https-portal_1  | 2019/05/30 23:50:04 [notice] 178#178: signal process started
https-portal_1  | [cont-init.d] 20-setup: exited 0.
https-portal_1  | [cont-init.d] 30-set-docker-gen-status: executing...
https-portal_1  | [cont-init.d] 30-set-docker-gen-status: exited 0.
https-portal_1  | [cont-init.d] done.
https-portal_1  | [services.d] starting services
https-portal_1  | [services.d] done.
```

I'm using Cloudflare and have the subdomain configured like so:

![image](https://user-images.githubusercontent.com/12100/58672318-422a9580-82fb-11e9-9151-f4be587162b6.png)

Can see it looks fine:

```
ping registry.tryorbit.com
PING registry.tryorbit.com (142.93.94.185): 56 data bytes
```

And matches to the droplet, which I can ssh into using `docker-machine ssh orbit-registry`.

I'm unable to curl the page when it does verification, but it goes so fast I don't think I'm checking it in time. I try and hit for example `http://registry.tryorbit.com/.well-known/acme-challenge/4yECewFte-w4NNPkG2b78gOcXvJCwVsW891fnoRGsG0` but by the time I copy/paste and run curl its either down already or never went up (I suspect).

I tried with a few configs including the simple one:

```
https-portal:
  image: steveltn/https-portal:1
  ports:
    - '80:80'
    - '443:443'
  environment:
    DOMAINS: 'registry.tryorbit.com'
```

But no dice. Even set up a new digitalocean server and ran it there on a new sub-domain, and get the same error. Sort of pulling my hair here... just earlier it was working fine. I didn't change anything in DNS, just did a down/up.

Edit with more info:

Checked Cloudflare just to be sure. There are no rewrites or firewall rules, I even turned of all SSL on the domain to be sure. Also ran `sudo ufw allow https` and `sudo ufw allow http` just in case on the server.

Ran this on the server to check 80 is open:

```
echo 'hi' > test.txt
{ echo -ne "HTTP/1.0 200 OK\r\n\r\n"; cat test.txt; } | nc -l -p 80
```

And was able to see it on registry.tryorbit.com just fine. So that looks to be working fine.

**Comment by SteveLTN** [comment]
Nginx doesn't know the verification fails or not, it won't stop serving `http://registry.tryorbit.com/.well-known/acme-challenge/4yECewFte-w4NNPkG2b78gOcXvJCwVsW891fnoRGsG0` after the registration fails.

Looks like port 80 isn't open to me.

**Comment by natew** [comment]
Did you see my edit above? I'm able to run `nc` and serve on port 80 just fine, so could this be docker not getting access somehow to 80 after a server restart? Thanks for the help, not much of a server/admin so stumbling through here.

**Comment by natew** [comment]
I'd be tempted to guess I just ran into rate limiting as I've tried it a ton of ways now, across a few servers, ips and different subdomains. But given I can run a local server on port 80 and curl it successfully, but then I run the docker image and curl the `http://registry.tryorbit.com/.well-known/acme-challenge/*` url while it's running and I too see connection refused, I'm not sure.

**Comment by natew** [comment]
Ok, so not sure what happened exactly. I realized that despite having the right docker env variables set, it seemed to not be running it on my remote machine but instead on my local one. Not sure this was even the fix as I tried a few things, but running a new shell, new docker-machine, and verifying with `docker run -d -p 8080:80 --name httpserver nginx` and then checking :8080 worked, and then it came up fine.
